### PR TITLE
Xpath scraper for goddesssnow.com

### PR DESCRIPTION
**Pull request by nocrad349**
Xpath scraper for goddesssnow.com

### CODE REVIEW

**Comment by JaseNZC** [comment]
Suggestion.
There is also a section under videos that could be scraped. [](urlhttps://goddesssnow.com/categories/movies.html
This covers that as well if you wanted to, killing two birds with one stone.

```yaml
name: insex
sceneByURL:
  - action: scrapeXPath
    url:
      - goddesssnow.com
    scraper: sceneScraper

xPathScrapers:
  sceneScraper:
    scene:
      Title: 
        selector: //div[@class="title_bar"]/span/text() | //h2[@class="update-title"]
      Date:
        selector: //span[@class="release-date"]/text() | //span[@class="update-date"]
        postProcess:
          - replace: 
            - regex: 'Release Date: ([0-9]{2}/[0-9]{2}/[0-9]{4})'
              with: $1
          - parseDate: 01/02/2006
      Details:
        selector: //span[@class="update_description"]/text() | //div[@class="update-description"]
      Performers: 
        Name: //span[@class="update_models"]/a/text() | //span[@class="tour_update_models"]/a
      Tags: 
        Name: //span[@class="update_tags"]/a/text() | //div[@class="update-tags"]/a
      Studio:
        Name: 
          fixed: Goddess Alexandra Snow
      Image: //meta[@name="twitter:image"]/@content

# Last Updated June 06, 2021
```

**Comment by nocrad349** [comment]
I thought about doing that, It's just that the non-vod pages you're talking about don't have the complete description/details, all descriptions end in an ellipsis.
Now that I think about it some more, the overall best solution would probably be using a subscraper, the non-vod pages have a link to the vod page. That will take me some time to figure out though, haven't don't anything with subscraping yet.

**Comment by JaseNZC** [comment]
Yup was just a suggestion :-) , I would not be too worried about the ellipsis it still gives a good description.

**Comment by nocrad349** [comment]
Using subScraper was easier than I thought, scraper now handles both parts of the site.

File: scrapers/goddesssnow.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+xPathScrapers:
+  vodScraper:
+    scene:
+      Title: 
+        selector: //div[@class="title_bar"]/span/text()
+      Date:
+        selector: //span[@class="release-date"]/text()
+        postProcess:
+          - replace: 
+            - regex: 'Release Date: ([0-9]{2}/[0-9]{2}/[0-9]{4})'
```
Comment (bnkai) [inline scrapers/goddesssnow.yml:21]: for `[0-9]` we use `\d` as it makess the regex a little cleaner
```yaml
              - regex: 'Release Date: (\d{2}/\d{2}/\d{4})'
 ```

File: scrapers/goddesssnow.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          fixed: Goddess Alexandra Snow
+      Image: &image
+        selector: //meta[@name="twitter:image"]/@content
+
+  updateScraper:
+    scene:
+      Title: 
+        selector: //h2[@class="update-title"]
+      Date:
+        selector: //span[@class="update-date"]
```
Comment (bnkai) [inline scrapers/goddesssnow.yml:41]: you forgot to parse the date
```yaml
      Date:
        selector: //span[@class="update-date"]
        postProcess:
          - parseDate: 01/02/2006
 ```

File: scrapers/goddesssnow.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      Details:
+        selector: //div[@class="update-join"]/a[2]/@href
+        postProcess:
+          - replace: 
+            - regex: (.*)
+              with: https://www.goddesssnow.com$1
```
Comment (bnkai) [inline scrapers/goddesssnow.yml:47]: to append at the start of a string we usually do
```yaml
              - regex: ^
                with: https://www.goddesssnow.com
```

**Pull request merged**